samples/sensor/grow_r502a/sample.yaml

```diff
@@ -11,4 +11,5 @@ tests:
     harness: sensor
     build_only: true
     depends_on: serial
-    filter: dt_compat_enabled("hzgrow,r502a")
+    platform_allow:
```

Comment:
Why change this from dt_compat to platform_allow? Seeing this a few times now

Reply:
Yeah I get it now. Avoiding DTS makes sense to me with this insight.

Reply:
Never thought about that. Might this information be worth adding to https://docs.zephyrproject.org/latest/develop/test/twister.html?